### PR TITLE
Sync docs from README.md to docs/*.md

### LINKED ISSUE #2825: Generate the options part of README.md and options.md from code
From https://github.com/prettier/prettier/pull/2767#issuecomment-327744705:

> Probably something for a future PR, but I've been thinking about generating the options part of README.md and options.md (which is probably out of date) from code.

Possible solution: 
- [markdown-magic](https://github.com/DavidWells/markdown-magic)

### PR DESCRIPTION
**Pull request by azz**
This is a lower-tech variant of #2825.

* Copies content from the main README.md file to the docs directory using `markdown-magic`. 
* Validated in CI via git diff.
* Replaces existing `markdown-toc` with `markdown-magic`

Closes #2825
Supersedes #2937

### CODE REVIEW

**Comment by azz** [comment]
Thoughts on whether we should generate from `README.md` to `docs/` or the other way around? Usually build steps are compositional rather than extracting bits, but people landing on the repo will probably submit edits to README and it would be strange to have part of that file generated, half hand-written.

I've redirected the <kbd>EDIT THIS DOC</kbd> buttons on the website to the `README.md` so we don't get people accidentally submitting changes to the docs files.

**Comment by duailibe** [comment]
Or maybe we should not have so much info in the README and link to docs instead? I know this might be controversial but I think it’s more sane

**Comment by azz** [comment]
Yeah that's also an option. We'd have to make sure everything in the README is covered elsewhere though. [Babel's README](https://github.com/babel/babel) works like this.

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "jest-docblock": "21.3.0-beta.1",
     "jest-validate": "21.1.0",
     "leven": "2.1.0",
+    "markdown-magic": "^0.1.19",
```
Comment (ikatyang) [inline package.json]: pin?

Comment (azz) [reply package.json]: Gah, I keep doing this. Need to figure out why my yarn isn't respecting `.yarnrc`.

**Comment by azz** [comment]
@vjeux thoughts on this vs just redirecting from Readme to website and building that out more?

**Comment by vjeux** [comment]
I don't feel strongly about the two variants. Either we redirect to the website or we make sure the two are kept in sync.

**Comment by azz** [comment]
Seems more sane to only keep the docs in one place. Better for maintenance and contribution via GitHub (without needing to run build tools). Opened #3139

**Pull request closed without merging**